staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -112,9 +116,50 @@ func (wm watchersMap) terminateAll(done func(*cacheWatcher)) {
 	}
 }
 
+type bucketWatchersMap map[int64][]*cacheWatcher
```

Comment:
do we expect this map to be used in other case? if not, probably just make this a timeBucketWatchMap